pylxd/models/storage_pool.py

```diff
@@ -466,10 +475,12 @@ def create(cls, storage_pool, *args, **kwargs):
         assert isinstance(definition, dict)
         assert "name" in definition
         response = storage_pool.api.volumes.custom.post(json=definition)
+        response_json = response.json()
 
-        if response.json()["type"] == "async" and wait:
+        # Handle both sync and async responses for endpoint changing from sync to async
+        if response_json["type"] == "async" and wait:
```

Comment:
Good catch! As another optimization we could do the JSON parsing only `if wait:`. I've done something similar in https://github.com/canonical/pylxd/pull/678